### PR TITLE
KFSPTS-14682 add processing for Proposals to the rass job

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Comment by db483** [comment]
retest this please

File: src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	@Override
+	public Proposal findExistingObject(RassXmlAwardEntry xmlAward) {
+		if (StringUtils.isBlank(xmlAward.getProposalNumber())) {
+			throw new RuntimeException("Attempted to search for a Proposal with a blank Proposal Number");
+		}
+		return getProposalByPrimaryKey(xmlAward.getProposalNumber());
+	}
+
+	private Proposal getProposalByPrimaryKey(String proposalNumber) {
+		Map<String, String> fields = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java]: I think in this case we just want ti know if a Proposal exists no matter what the active status is. I checked the Ezra batch job  and that works the same.

File: src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		return proposal;
+	}
+
+	@Override
+	public boolean businessObjectEditIsPermitted(RassXmlAwardEntry xmlObject, Proposal oldBusinessObject) {
+		return false;
+	}
+
+	@Override
+	public void processCustomTranslationForBusinessObjectEdit(RassXmlAwardEntry xmlObject, Proposal oldBusinessObject,
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java]: I will remove it

File: src/main/java/edu/cornell/kfs/rass/batch/RassOrganizationConverter.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package edu.cornell.kfs.rass.batch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.kuali.kfs.module.cg.businessobject.ProposalOrganization;
+
+public class RassOrganizationConverter extends RassValueConverterBase {
+	private static final String PROPOSAL_ORG_CHART = "IT";
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/RassOrganizationConverter.java]: I will move it

File: src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java [hunk 4]
```diff
@@ -374,6 +403,19 @@ protected String cleanStringValue(
         }
         return cleanedValue;
     }
+    
+    protected java.sql.Date cleanDateValue(
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassServiceImpl.java]: this is just returning a java.sql.Date from a java.util.Date

File: src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java]: OK

**Comment by db483** [comment]
This is ready for review again.

File: src/main/java/edu/cornell/kfs/rass/batch/RassProjectDirectorConverter.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+public class RassProjectDirectorConverter extends RassValueConverterBase {
+
+	@Override
+	public Object convert(Object rassInput) {
+		List<RassXMLAwardPiCoPiEntry> rassAwardPiCoPiEntries = (List<RassXMLAwardPiCoPiEntry>) rassInput;
+		List<ProposalProjectDirector> projectDirectors = new ArrayList<>();
+		if (rassAwardPiCoPiEntries != null && rassAwardPiCoPiEntries.size() > 0) {
+			for (RassXMLAwardPiCoPiEntry rassAwardPiCoPi : rassAwardPiCoPiEntries) {
+				ProposalProjectDirector projectDirector = new ProposalProjectDirector();
+				String principalId = SpringContext.getBean(PersonService.class)
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/rass/batch/RassProjectDirectorConverter.java]: good point, making the change

**Comment by db483** [comment]
This is ready for review again.

**Pull request merged**